ci: "Create PR to increment version" step failing on v4.0.0 branch

### Description
The "Publish to npm" workflow on the `v4.0.0` branch is failing at the "Create PR to increment version" step.

**Logs:**
> Pushing pull request branch to 'origin/ap-publish-v3.25.7'
> Error: Action failed with error: Process completed with exit code 1

### Root Cause
The workflow is using `peter-evans/create-pull-request@v3`, which is outdated and no longer works with current GitHub Actions token permissions for pushing branches. The `main` branch has already been updated to `@v7`.

### Proposed Fix
Update `.github/workflows/publish.yml` on the `v4.0.0` branch to use `peter-evans/create-pull-request@v7`.

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -16,7 +16,7 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Use Node.js 20.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v4
         with:
           node-version: 20.x
 
@@ -35,7 +35,7 @@ jobs:
           npm publish --workspaces --access public ${{ steps.tag.outputs.tag }} 2>&1
 
       - name: Create PR to increment version
-        uses: peter-evans/create-pull-request@v3
+        uses: peter-evans/create-pull-request@v7
         with:
           base: main
           commit-message: 'chore(actions): publish ${{ github.event.release.tag_name }} to npm'

